### PR TITLE
Adds Discord provider.

### PR DESCRIPTION
**Pull request by VagantemNumen**
Adds Discord provider.

### CODE REVIEW

**Comment by markbates** [comment]
You want to leave the fields that aren't provider by Discord to be blank.

For example `user.Location`. By setting it equal to "No location is provided by the Discord API" then I can't check that it is empty and prompt the user to enter a location for me.

File: providers/discord/discord.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+	}{}
+
+	err := json.NewDecoder(r).Decode(&u)
+	if err != nil {
+		return err
+	}
+
+	user.Name = u.Name
+	user.Email = u.Email
+	user.NickName = "No nickname is provided by the Discord API"
```
Comment (markbates) [inline providers/discord/discord.go]: Leave blank

File: providers/discord/discord.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+	user.Location = "No location is provided by the Discord API"
```
Comment (markbates) [inline providers/discord/discord.go]: Leave blank

File: providers/discord/discord.go [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+	user.AvatarURL = "https://discordapp.com/api/users/" + u.ID + "/avatars/" + u.AvatarID + ".jpg"
+	user.UserID = u.ID
+	// user.Discriminator	= u.Discriminator
```
Comment (markbates) [inline providers/discord/discord.go]: remove commented code

**Comment by VagantemNumen** [comment]
By the way, what to do with fields provided by discord but not in the ``goth.User`` struct? Most of those fields are quite useful too.

**Comment by markbates** [comment]
You want to put those into the `user.RawData` field for people to access later.

**Comment by VagantemNumen** [comment]
Thank you for pointing that out. 
I somehow missed it.
Done.

**Pull request merged**